base/linalg/generic.jl

```diff
@@ -417,48 +417,38 @@ istriu(x::Number) = true
 istril(x::Number) = true
 isdiag(x::Number) = true
 
-#TODO: lookup where all the documentation for linreg is coming from, including the example
-# mentioning PyPlot etc
 """
     linreg(x, y)
 
 Find the Ordinary Least Squares best linear fit between the numerical vectors `x` and `y`
 
 Inputs:
 
-- `x::AbstractVector{T <: Number}` - vector of independent variables
-- `y::AbstractVector{S <: Number}` - vector of dependent variables
+- `x::AbstractVector` - vector of independent variables
+- `y::AbstractVector` - vector of dependent variables
 
 Output:
 
-- `[b0, b1]` - the coefficients for the line of best fit such that yhat = b0 + b1*x
+- `(a, b)` - the coefficients for the line of best fit satisfying a + b*x
 
 See also:
 
 `\\`, `cov`, `std`, `mean`
 
 """
-function linreg{T <: Number, S <: Number}(x::AbstractVector{T}, y::AbstractVector{S})
+function linreg(x::AbstractVector, y::AbstractVector)
     # Least squares given
-    # Y = b0 + b1*X
+    # Y = a + b*X
     # where
-    # b1 = cov(X, Y)/var(X)
-    # b0 = mean(Y) - b1*mean(X)
-    # Calculate the cov and var using the fewest iterations of the inputs possible
-    n = length(x)
-    @assert n == length(y)
-    mx = mean(x) # need to know this at each step do the implicit loop over x first
-    tot_dx = 0.0
-    tot_y = 0.0
-    tot_xy = 0.0
-    for i in 1:n
-        tot_dx += (x[i] - mx)^2
-        tot_y += y[i]
-        tot_xy += x[i]*y[i]
-    end
-    b1 = (tot_xy - mx*tot_y)/tot_dx # a 1\n cancels in the top and bottom
-    b0 = tot_y/n - b1*mx
-    return [b0, b1]
+    # b = cov(X, Y)/var(X)
+    # a = mean(Y) - b*mean(X)
+    size(x) == size(y) || throw(DimensionMismatch())
```

Comment:
I will look into this. Was just following the code from the recent changes to `quantile`